### PR TITLE
Remove legacy script code paths

### PR DESCRIPTION
**Pull request by joshlory**
The change cleans up old code paths associated with legacy default/*.yml scripts.

### CODE REVIEW

**Comment by wjordan** [comment]
:+1: another very nice, thorough refactor that will make this code so much easier to work with!

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -112,25 +100,17 @@ def twenty_hour?
 
   def hoc?
     # note that now multiple scripts can be an 'hour of code' script
-    self.id == HOC_ID || self.name == HOC_NAME || self.name == FROZEN_NAME || self.flappy? || self.name == PLAYLAB_NAME
+    self.name == HOC_2013_NAME || self.name == HOC_NAME || self.name == FROZEN_NAME || self.flappy? || self.name == PLAYLAB_NAME
```
Comment (laurelfan) [inline dashboard/app/models/script.rb]: could change this to a [].include? like above

File: dashboard/app/models/script.rb [hunk 2]
```diff
@@ -183,13 +163,9 @@ def has_lesson_plan?
   SCRIPT_CSV_MAPPING = %w(Game Name Level:level_num Skin Concepts Url:level_url Stage)
   SCRIPT_MAP = Hash[SCRIPT_CSV_MAPPING.map { |x| x.include?(':') ? x.split(':') : [x, x.downcase] }]
 
-  def self.setup(default_files, custom_files)
+  def self.setup(custom_files)
```
Comment (laurelfan) [inline dashboard/app/models/script.rb]: :+1:

File: dashboard/app/views/shared/_course_big_block.haml [hunk 3]
```diff
@@ -5,7 +5,7 @@
   - image_url = image_tag('signup_header_cropped.jpg')
 
 - elsif id == 'hoc'
-  - url = hoc_reset_url  # script_url(Script::HOC_ID)
+  - url = hoc_reset_url  # script_url(Script::HOC_2013_NAME)
```
Comment (laurelfan) [inline dashboard/app/views/shared/_course_big_block.haml]: not sure this comment means anything

**Comment by laurelfan** [comment]
thanks for taking on all this refactoring, and yay for removing functions with _or_ in the name :)

**Pull request merged**